bondcpp/src/bond.cpp

```diff
@@ -516,38 +539,36 @@ void Bond::bondStatusCB(const bond::msg::Status::ConstSharedPtr msg)
 
 void Bond::doPublishing()
 {
-  std::unique_lock<std::mutex> lock(mutex_);
-  if (sm_.getState().getId() == SM::WaitingForSister.getId() ||
-    sm_.getState().getId() == SM::Alive.getId())
-  {
+  std::unique_lock<std::mutex> lock(state_machine_mutex_);
```

Comment:
I think we need to remove this line now, right?  Otherwise we'll deadlock since `isStateWaitingForSister`, `isStateAlive`, and `isStateAwaitSisterDeath` also take this lock.

Reply:
Yep.  That is what is causing CI to fail.